addons/hr_holidays/models/hr_leave.py

```diff
@@ -1438,6 +1441,48 @@ def message_subscribe(self, partner_ids=None, subtype_ids=None):
             return super(HrLeave, self.sudo()).message_subscribe(partner_ids=partner_ids, subtype_ids=subtype_ids)
         return super().message_subscribe(partner_ids=partner_ids, subtype_ids=subtype_ids)
 
+    def _notify_leave_request(self):
+        """Notifies the responsible approver(s) when an employee applies for leave."""
+        if self.env.context.get('mail_activity_automation_skip'):
+            return False
+
+        locale = get_lang(self.env).code
+        ticket_model_description = self.env['ir.model']._get(self._name).display_name
+        subject = self.env.ref('hr_holidays.new_timeoff_request_template')._render_field('subject', self.ids, compute_lang=True)
+        for holiday in self:
+            if holiday.holiday_status_id.leave_validation_type == 'no_validation':
+                continue
+
+            partner_ids = holiday.sudo()._get_responsible_for_approval().mapped('partner_id')
+
+            if not partner_ids:
+                continue
+            leave_msg = self.env.ref('hr_holidays.new_timeoff_request_template').with_context(
+                manager=', '.join(partner_ids.mapped('name'))
+            )._render_field(
+                'body_html',
+                self.ids,
+                compute_lang=True
+            )[holiday.id]
+
+            subtitles = [
+                _('%(employee_name)s requested a time-off \n',
+                    employee_name=holiday.employee_id.name),
+                _('%(start_date)s → %(end_date)s (%(duration)s)',
+                    start_date=babel.dates.format_date(holiday.request_date_from, 'MMM dd', locale=locale),
+                    end_date=babel.dates.format_date(holiday.request_date_to, 'MMM dd', locale=locale),
```

Comment:
Use ``format_date`` instead of babel, we want the dates to be localized according to the employee environment.